syntax = "proto3"; 

package business.business_service; 

import "error/user_error.proto";

message CreateBusinessRequest {
    message Business {
        string business_name = 1;
        string email = 2;
        string website_url = 3;
        string phone_number = 4;
        string location_city = 5;
        string location_state = 6;
        string ein = 7;
        string business_description = 8;
        string business_size = 9;
    }
    Business business = 1;
    string user_email = 2;  // Email of the user creating the business
} 

message CreateBusinessResponse { 
    message Business {
        string id = 1;
        string business_name = 2;
        string email = 3;
        string website_url = 4;
        string phone_number = 5;
        string location_city = 6;
        string location_state = 7;
        string ein = 8;
        string business_description = 9;
        string business_size = 10;
    }
    Business business = 1;
    repeated user.error.UserError errors = 2;
}
